specification/scvmm/resource-manager/Microsoft.ScVmm/stable/2024-06-01/scvmm.json

```diff
@@ -4174,6 +4186,41 @@
           "type": "string",
           "description": "Gets os version.",
           "readOnly": true
+        },
+        "domainName": {
+          "type": "string",
+          "description": "Gets or sets the domain name."
+        },
+        "domainUsername": {
+          "type": "string",
+          "description": "Gets or sets the domain username."
+        },
+        "domainPassword": {
+          "type": "string",
+          "format": "password",
+          "description": "Password of the domain the VM has to join.",
+          "x-ms-mutability": [
+            "update",
+            "create"
+          ],
+          "x-ms-secret": true
+        },
+        "workgroup": {
+          "type": "string",
+          "description": "Gets or sets the workgroup."
+        },
+        "productKey": {
+          "type": "string",
+          "description": "Gets or sets the product key.Input format xxxxx-xxxxx-xxxxx-xxxxx-xxxxx"
+        },
+        "timezone": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Gets or sets the index value of the timezone."
+        },
+        "runOnceCommands": {
```

Comment:
If these commands are run only during creation, the values will be persisted inside object forever? What would be the behavior during update operation?